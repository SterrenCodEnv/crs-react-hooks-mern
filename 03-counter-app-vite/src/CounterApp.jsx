import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleIncrease = () => {
    setCounter(counter + 1);
  };

  const handleDecrease = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>

      <button onClick={handleIncrease}> +1 </button>
      <button onClick={handleDecrease}> -1 </button>
      <button aria-label="btn-reset" onClick={handleReset}>
        {' '}
        Reset{' '}
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
