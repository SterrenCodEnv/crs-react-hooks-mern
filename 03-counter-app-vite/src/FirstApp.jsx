import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, author }) => {
  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      <p>{subTitle}</p>
      <p>{author}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  title: 'No Title',
  subTitle: 'No SubTitle',
  author: 'Admin',
};
