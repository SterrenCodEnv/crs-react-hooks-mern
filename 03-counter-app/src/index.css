html, body{
    background-color: #21232A;
    color: white;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    font-size: 1.2rem;
    padding: 70px;
}

button{
    font-size: 1.2rem;
    padding: 10px;
    margin-right: 10px;
}