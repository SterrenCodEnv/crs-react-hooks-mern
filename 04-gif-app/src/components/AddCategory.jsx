import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target: { value } }) => {
    setInputValue(value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.length >= 1) {
      onNewCategory(inputValue.trim().toLocaleLowerCase());
      setInputValue('');
    }
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        onChange={onInputChange}
        value={inputValue}
        type="text"
        placeholder="Buscar gif..."
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
