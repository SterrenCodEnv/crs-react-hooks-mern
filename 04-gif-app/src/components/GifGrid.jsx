import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <div key={category}>
        <h2>{`${
          category.charAt(0).toUpperCase() + category.slice(1).toLowerCase()
        }`}</h2>

        {isLoading && <h2>Loading...</h2>}

        <div className="card-grid">
          {images.map((image) => (
            <GifItem key={image.id} {...image} />
          ))}
        </div>
      </div>
    </>
  );
};
