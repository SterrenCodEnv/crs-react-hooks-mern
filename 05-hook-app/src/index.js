import React from "react";
import ReactDOM from "react-dom";
import { Father } from "./components/07-MemoTask/Father";
// import { CallbackHook } from "./components/06-Memos/CallbackHook";
// import { MemoHook } from "./components/06-Memos/MemoHook";
// import { Memorizes } from "./components/06-Memos/Memorizes";
// import { Layout } from "./components/05-useLayoutEffect/Layout";
// import { RealExampleRef } from "./components/04-userRef/RealExampleRef";
// import { FocusScreen } from "./components/04-userRef/FocusScreen";
// import { MultipleCustomHooks } from "./components/03-Examples/MultipleCustomHooks";
// import { FormWithCustomHook } from "./components/02-useEffect/FormWithCustomHook";
// import { SimpleForm } from "./components/2-useEffect/SimpleForm";
// import { CounterWithCustomHook } from "./components/01-useState/CounterWithCustomHook";
// import { CounterApp } from "./components/01-useState/CounterApp";
// import { HookApp } from "./HookApp";

ReactDOM.render(
  //<HookApp />,
  // <CounterWithCustomHook />,
  // <SimpleForm />,
  // <FormWithCustomHook />,
  // <MultipleCustomHooks />,
  // <FocusScreen />,
  // <RealExampleRef />,
  // <Layout />,
  // <Memorizes />,
  // <MemoHook />,
  // <CallbackHook />,
  <Father />,
  document.getElementById("root")
);
