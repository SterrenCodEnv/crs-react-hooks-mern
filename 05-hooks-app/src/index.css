body {
    padding: 20px;
}

button {
    margin-right: 10px;
}

.btn-circle.btn-xl {
    width: 70px;
    height: 70px;
    padding: 10px 16px;
    border-radius: 35px;
    font-size: 24px;
    line-height: 1.33;
}

.btn-circle {
    width: 30px;
    height: 30px;
    padding: 6px 0px;
    border-radius: 15px;
    text-align: center;
    font-size: 12px;
    line-height: 1.42857;
}

.unselect {
    -webkit-touch-callout: none; /* iOS Safari */
      -webkit-user-select: none; /* Safari */
       -khtml-user-select: none; /* Konqueror HTML */
         -moz-user-select: none; /* Old versions of Firefox */
          -ms-user-select: none; /* Internet Explorer/Edge */
              user-select: none; /* Non-prefixed version, currently
                                    supported by Chrome, Edge, Opera and Firefox */
  }