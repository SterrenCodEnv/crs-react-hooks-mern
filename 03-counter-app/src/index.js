import React from 'react';
import ReactDOM from 'react-dom'
// import FirstApp from './FirstApp';
import CounterApp from './CounterApp';
import './index.css'

const divRoot = document.querySelector('#app');

// ReactDOM.render( <FirstApp portMessage='Listening on port' />, divRoot);
ReactDOM.render( <CounterApp year={2021} />, divRoot);