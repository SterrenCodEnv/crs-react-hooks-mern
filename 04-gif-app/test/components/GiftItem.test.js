import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('GifItem Component Test', () => {
  const title = 'Gif Title';
  const url =
    'https://api.giphy.com/v1/gifs/search?api_key=0kePs8YM92zgFmRSM7xoOpXX8QJ6mUFr&q=puppies&limit=20';

  const component = <GifItem title={title} url={url} />;

  test('Should must match snapshot ', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  test('Should show image with url and alt', () => {
    render(component);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Should show title into component', () => {
    render(component);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
