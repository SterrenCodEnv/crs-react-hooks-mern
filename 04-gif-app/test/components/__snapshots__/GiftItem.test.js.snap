// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Component Test Should must match snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif Title"
      src="https://api.giphy.com/v1/gifs/search?api_key=0kePs8YM92zgFmRSM7xoOpXX8QJ6mUFr&q=puppies&limit=20"
    />
    <p>
      Gif Title
    </p>
  </div>
</div>
`;
